### PR TITLE
Sanitize `ImageDimensions` base class

### PR DESCRIPTION
**Pull request by Floessie**
Hi there, post-5.0 developers, :smile: 

Today I was browsing the code to find out, why we need virtual inheritance. Now, that point is clear to me and not that easy to change (well, you know, virtual inheritance has a certain smell...). Nevertheless I took the opportunity to refactor the very base class `ImageDimensions` with these goals in mind:

- Make `width` and `height` private
- Drop `getW()` and `getH()`
- Clean `PreviewProps`

I hope you like the result.

Best
Flössie

### CODE REVIEW

**Comment by Beep6581** [comment]
Hey @Floessie, how do I test this? Or maybe just merge and deal with eventual fallout? Remember to change `master` to `dev`.

**Comment by Floessie** [comment]
@Beep6581 I pushed a [rebased branch](https://github.com/Beep6581/RawTherapee/tree/imdim_cleanup-dev) for you to test.

> Hey @Floessie, how do I test this? Or maybe just merge and deal with eventual fallout?

It's purely code cosmetic. The only thing that could have been broken is `ImageDimensions::transform()`, which is called by `StdImageSource::getImage()` and deals with rotation (90, 180, 270) and flip.

HTH,
Flössie

File: rtengine/imagedimensions.cc [hunk 1]
```diff
@@ -20,49 +20,88 @@
 #include "imagedimensions.h"
 #include "rtengine.h"
 
-void ImageDimensions::transform (PreviewProps pp, int tran, int &sx1, int &sy1, int &sx2, int &sy2)
+PreviewProps::PreviewProps(int _x, int _y, int _width, int _height, int _skip) :
```
Comment (Hombre57) [inline rtengine/imagedimensions.cc:23]: You do not need to add underscore for local variables, `x(x)` is a valid syntax and already used several times in RT.

File: rtengine/imagedimensions.h [hunk 2]
```diff
@@ -17,47 +17,47 @@
...
-#define _IMAGEDIMENSIONS_
+#pragma once
 
 class PreviewProps
 {
 public:
-    int x, y, w, h, skip;
-    PreviewProps (int _x, int _y, int _w, int _h, int _skip)
-        : x(_x), y(_y), w(_w), h(_h), skip(_skip) {}
+    PreviewProps(int _x, int _y, int _width, int _height, int _skip);
```
Comment (Hombre57) [inline rtengine/imagedimensions.h:25]: Ok, I see that it already used underscore before. Nevermind.

**Comment by Hombre57** [comment]
+1 to commit. Thanks for the cleanup !

**Comment by Floessie** [comment]
@Hombre57 My pleasure! Merged with 6daf5f6011ac8f15ac1e743f622efe87c8652983.

**Pull request closed without merging**